library/std/src/keyword_docs.rs

```diff
@@ -14,6 +14,16 @@
 /// assert_eq!(true as u8 + thing2 as u8, 100);
 /// ```
 ///
+/// `as` can also be used with the `_` placeholder when the destination type
+/// can be inferred. Note that this can cause inference breakage and usually
+/// such code should use an explicit type for both clarity and stability:
+///
+/// ```rust
+/// let value1: i32 = 42;
+/// let value2: i64 = value1 as _;
+/// assert_eq!(value2, value1 as _);
+/// ```
```

Comment:
I feel like this is ambiguous whether this is actually providing an example of "good" or "bad" code. Maybe we can adjust the wording to be clearer, or just not providing a code example here is fine -- we don't want to recommend the pattern after all :)